components/ArticleItemWidget/ArticleItemWidget.js

```diff
@@ -0,0 +1,255 @@
+};
+
+export default class ArticleItemWidget extends PureComponent {
+  static propTypes = {
+    id: PropTypes.string.isRequired, // Article ID
+    read: PropTypes.bool.isRequired,
+    notArticleReplied: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.string, // TYPE_SUGGESTION_OPTIONS in replyType.js
+    ]),
+    handleLocalEditorHelperList: PropTypes.func.isRequired,
+  };
+
+  replyToNotArticle = text => {
+    const { id } = this.props;
+    NProgress.start();
+    gql`
+      mutation(
+        $articleId: String!
+        $text: String!
+        $type: ReplyTypeEnum!
+        $reference: String
+      ) {
+        CreateReply(
+          articleId: $articleId
+          text: $text
+          type: $type
+          reference: $reference
+        ) {
+          id
+        }
+      }
+    `({ articleId: id, type: 'NOT_ARTICLE', text }).then(() => {
+      this.handleNotArticleReply(text);
+      this.resetWidgetUI();
+      NProgress.done();
+    });
+  };
+
+  // the tags config
+  tags = [
+    {
+      name: 'read',
+      description: '已讀',
+      color: '#cfcfcf',
+      onClick: event => {
+        event && event.preventDefault();
+        this.handleRead();
+        this.resetWidgetUI();
+      },
+    },
+    {
+      name: 'notArticle',
+      description: '非查證',
+      color: 'orange',
+      extendComponent: (
+        <NotArticleExtendSelection replyToNotArticle={this.replyToNotArticle} />
+      ),
+      onClick: () => {
+        this.setState({
+          notArticleSelectionDisplay: !this.state.notArticleSelectionDisplay,
+        });
+      },
+    },
+  ];
+
+  constructor(props) {
+    super(props);
+
+    this.initialTagsState = {};
+    this.tags.forEach(section => {
+      this.initialTagsState[`${section.name}Hover`] = false;
+    });
+
+    this.state = Object.assign({}, initialState, this.initialTagsState);
+  }
+
+  componentDidMount = () => {
+    this.preventMobileLongPressMenu();
+  };
+
+  preventMobileLongPressMenu = () => {
+    this.refWidget.addEventListener('contextmenu', event =>
+      event.preventDefault()
+    );
+  };
+
+  preventAll = event => {
+    // stop event cascade (touchstart → touchend → mouseover → mousemove → mousedown → mouseup → click)
+    event.preventDefault();
+    // stop (child touchstart -> parent touchstart,... ) mainly for for parent a:link redirection,
+    event.stopPropagation();
+    return event;
+  };
+
+  // close widget bar and tags extend component
+  resetWidgetUI = () => {
+    this.setState(Object.assign({}, initialState, this.initialTagsState));
+  };
+
+  openBar = () => {
+    this.setState({
+      barOpen: true,
+      readHover: true, //for mobile, explain in onTouchMoveTip function
+      touchTarget: 'read',
+    });
+  };
+
+  hoverTag = event => {
+    const name = event.currentTarget.attributes['name'].value;
+    this.setState({
+      [`${name}Hover`]: true,
```

Comment:
Is it possible to hover multiple items at once?

If not, we should just go for one single "hoverIdx" in state, instead of multiple `xxxHover`s. This way we can get rid of `this.initialTagsState`.